Framework/Algorithms/src/XrayAbsorptionCorrection.cpp

```diff
@@ -0,0 +1,234 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "MantidAlgorithms/XrayAbsorptionCorrection.h"
+#include "MantidAPI/Axis.h"
+#include "MantidAPI/Sample.h"
+#include "MantidAPI/SpectrumInfo.h"
+#include "MantidAPI/WorkspaceUnitValidator.h"
+#include "MantidKernel/BoundedValidator.h"
+#include "MantidKernel/CompositeValidator.h"
+#include "MantidKernel/Material.h"
+#include "MantidKernel/PhysicalConstants.h"
+#include "MantidKernel/VectorHelper.h"
+#include <iostream>
+#include <math.h>
+#include <numeric>
+
+using namespace Mantid::API;
+using namespace Mantid::Geometry;
+using namespace Mantid::Kernel;
+using Mantid::DataObjects::Workspace2D;
+namespace PhysicalConstants = Mantid::PhysicalConstants;
+
+/// @cond
+namespace {
+
+constexpr double DEFAULT_ANGLE = 45.0;
```

Comment:
units?

Reply:
Supposed to be degrees will put that in a comment